src/transformers/models/resnetd/convert_resnetd_to_pytorch.py

```diff
@@ -0,0 +1,215 @@
+        x = torch.randn((1, 3, 224, 224))
+        module_transfer(x)
+
+    from_logits = from_model(x)
+    our_logits = our_model(x).logits
+    assert torch.allclose(from_logits, our_logits), f"The model {name} logits don't match the original one."
+
+    checkpoint_name = name
+    print(checkpoint_name)
+
+    if push_to_hub:
+        our_model.push_to_hub(
+            repo_path_or_name=save_directory / checkpoint_name,
+            commit_message="Add model",
+            use_temp_dir=True,
+        )
+
+        # we can use the convnext one
+        feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/convnext-base-224-22k-1k")
+        feature_extractor.push_to_hub(
+            repo_path_or_name=save_directory / checkpoint_name,
+            commit_message="Add feature extractor",
+            use_temp_dir=True,
+        )
+
+        print(f"Pushed {checkpoint_name}")
+
+
+def convert_weights_and_push(save_directory: Path, model_name: str = None, push_to_hub: bool = True):
+    filename = "imagenet-1k-id2label.json"
+    num_labels = 1000
+    expected_shape = (1, num_labels)
+
+    repo_id = "datasets/huggingface/label-files"
+    num_labels = num_labels
+    id2label = json.load(open(cached_download(hf_hub_url(repo_id, filename)), "r"))
+    id2label = {int(k): v for k, v in id2label.items()}
+
+    id2label = id2label
+    label2id = {v: k for k, v in id2label.items()}
+
+    ImageNetPreTrainedConfig = partial(ResNetDConfig, num_labels=num_labels, id2label=id2label, label2id=label2id)
+
+    names_to_config = {
+        # "resnetd-18": ImageNetPreTrainedConfig(
+        #     depths=[2, 2, 2, 2], hidden_sizes=[64, 128, 256, 512], layer_type="basic"
+        # ),
+        # "resnetd-26": ImageNetPreTrainedConfig(
+        #     depths=[2, 2, 2, 2], hidden_sizes=[256, 512, 1024, 2048], layer_type="bottleneck"
+        # ),
+        # "resnetd-34": ImageNetPreTrainedConfig(
+        #     depths=[3, 4, 6, 3], hidden_sizes=[64, 128, 256, 512], layer_type="basic"
+        # ),
+        # "resnetd-50": ImageNetPreTrainedConfig(
+        #     depths=[3, 4, 6, 3], hidden_sizes=[256, 512, 1024, 2048], layer_type="bottleneck"
+        # ),
+        # "resnetd-101": ImageNetPreTrainedConfig(
+        #     depths=[3, 4, 23, 3], hidden_sizes=[256, 512, 1024, 2048], layer_type="bottleneck"
+        # ),
+        # "resnetd-152": ImageNetPreTrainedConfig(
+        #     depths=[3, 8, 36, 3], hidden_sizes=[256, 512, 1024, 2048], layer_type="bottleneck"
+        # ),
+        "resnetd-200": ImageNetPreTrainedConfig(
+            depths=[3, 24, 36, 3], hidden_sizes=[256, 512, 1024, 2048], layer_type="bottleneck"
+        ),
+    }
+
+    names_to_timm = {
+        # "resnetd-18": "resnet18d",
+        # "resnetd-26": "resnet26d",
+        # "resnetd-34": "resnet34d",
+        # "resnetd-50": "resnet50d",
+        # "resnetd-101": "resnet101d",
+        # "resnetd-152": "resnet152d",
+        "resnetd-200": "resnet200d"
```

Comment:
Same here? Also is it not easier to just remove the `-` amd move the d at the end?

Reply:
I've change the names to `resnet-d-***` (similar to `regnet-x/y-***`)